### ISSUE TITLE
🐛 BUG: LS crashes on neovim

### ISSUE THREAD

**Issue by aynp** [withastro/language-tools#730]
### Describe the Bug

LS crashes and fails to attach on neovim.

###  Log 
```
[ERROR][2023-12-20 03:57:21] .../vim/lsp/rpc.lua:734"rpc""astro-ls""stderr""/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/dist/languageServerPlugin.js:45
                readDirectory: modules.typescript.sys.readDirectory,
                                                  ^

TypeError: Cannot read properties of undefined (reading sys)
    at Object.resolveConfig (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/dist/languageServerPlugin.js:45:51)
    at createProject (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/project.js:78:35)
    at async getParsedCommandLine (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/workspaces.js:302:29)
    at async prepareClosestootParsedCommandLine (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/workspaces.js:224:17)
    at async findMatchConfigs (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/workspaces.js:213:9)
    at async getProjectAndTsConfig (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/workspaces.js:154:30)
    at async sendDocumentDiagnostics (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/workspaces.js:136:26)
    at async updateDiagnostics (/home/aryan/.local/share/nvim/mason/packages/astro-language-server/node_modules/@astrojs/language-server/node_modules/@volar/language-server/out/common/workspaces.js:125:13)

Node.js v21.4.0
```

- On VS Code, using the official astro extension, LSP works fine.

### Steps to Reproduce

1. `pnpm create astro@latest`
2. Install LS using npm or a manager like Mason (tried both with the same result)
3. Setup LS ([here](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#astro))
4. Open an astro file.
5. LS crashes instead of attaching normally.

Sorry if out of scope. Please feel free to close if so.

**Comment by Princesseuh** [comment]
I think you might be missing some configuration, namely the path to TypeScript in your init config, but I don't really use Neovim so I'm not sure.

Nonetheless, I believe this crash has been fixed by https://github.com/withastro/language-tools/pull/701 already, hopefully it gives a better error message once that lands.

**Comment by aynp** [comment]
Sorry for duplicate issue.
Thanks a lot for pointing it out and fix.

As a reference here's how to add missing path to ts for now (it uses the typescript in the project).
```lua
require("lspconfig").astro.setup {
  init_options = {
    typescript = {
      tsdk = 'node_modules/typescript/lib'
    }
  }
}
```
